### PR TITLE
Add stubs for hdbcli

### PR DESCRIPTION
**Pull request by kasium**
hdbcli is the python dbapi for SAP HANA

### CODE REVIEW

**Comment by kasium** [comment]
I get the following errors,  which I don't get
```
stubs/hdbcli/hdbcli/dbapi.pyi:46: error: Unexpected "..."  [misc]
stubs/hdbcli/hdbcli/dbapi.pyi:73: error: An implementation for an overloaded function is not allowed in a stub file  [misc]
```

File: stubs/hdbcli/hdbcli/dbapi.pyi [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    def setautocommit(self, auto: bool = ...) -> None: ...
+    def setclientinfo(self, key: str, value: str | None = ...) -> None: ...
+
+class LOB:
+    def close(self) -> bool: ...
+    def find(self, object: str, length: int, position: int = ...) -> int: ...
+    def read(self, size: int = ..., position: int = ...) -> str | bytes: ...
+    def write(self, object: str | bytes) -> int: ...
+
+_Parameters = Sequence[tuple[Any, ...]]
```
Comment (JelleZijlstra) [inline stubs/hdbcli/hdbcli/dbapi.pyi:46]:
```suggestion
_Parameters = Sequence[Tuple[Any, ...]]
```
The current release of mypy doesn't support lowercase `tuple` in this context yet. I think there are a few other instances in this file.

Comment (JelleZijlstra) [reply stubs/hdbcli/hdbcli/dbapi.pyi:46]: This might only affect type aliases, you'll have to check.

File: stubs/hdbcli/hdbcli/dbapi.pyi [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    def parameter_description(self) -> Tuple[str, ...]: ...
+    @overload
+    def prepare(self, operation: str, newcursor: Literal[True]) -> Cursor: ...
+    @overload
+    def prepare(self, operation: str, newcursor: Literal[False]) -> Any: ...
+    def scroll(self, value: int, mode: Literal["absolute"] | Literal["relative"] = ...) -> None: ...
+    def server_cpu_time(self) -> int: ...
+    def server_memory_usage(self) -> int: ...
+    def server_processing_time(self) -> int: ...
+    def setinputsizes(self, *args: Any, **kwwargs: Any) -> None: ...
```
Comment (srittau) [inline stubs/hdbcli/hdbcli/dbapi.pyi:79]: Nit: Is the argument really called `kwwargs` (double-w)?

File: stubs/hdbcli/hdbcli/dbapi.pyi [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    def setfetchsize(self, value: int) -> None: ...
+    def set_resultset_holdability(self, holdability: int) -> None: ...
+    def setoutputsize(self, *args: Any, **kwwargs: Any) -> None: ...
```
Comment (srittau) [inline stubs/hdbcli/hdbcli/dbapi.pyi:82]: Same here.

**Review by srittau — APPROVED** [review]
Two nits (for future consideration), but LGTM.

File: stubs/hdbcli/hdbcli/dbapi.pyi [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+def TimestampFromTicks(ticks: float) -> datetime: ...
+def Binary(data: ReadableBuffer) -> memoryview: ...
+
+Decimal: Type[decimal.Decimal]
+
+NUMBER: Type[int] | Type[float] | Type[complex]
+DATETIME: Type[date] | Type[time] | Type[datetime]
+STRING: Type[str]
+BINARY: Type[memoryview]
+ROWID: Type[int]
```
Comment (srittau) [inline stubs/hdbcli/hdbcli/dbapi.pyi:115]: Also, did you try to use simple aliases here?

```suggestion
STRING = str
BINARY = memoryview
ROWID = int
```

**Pull request merged**